### PR TITLE
Fix bugs for alerting

### PR DESCRIPTION
**Pull request by zionwu**
Relate issues:
https://github.com/rancher/rancher/issues/12618, https://github.com/rancher/rancher/issues/12619, https://github.com/rancher/rancher/issues/13414

### CODE REVIEW

File: pkg/api/customization/alert/notifier_action.go [hunk 1]
```diff
@@ -306,20 +312,39 @@ func testEmail(host, password, username string, port int, requireTLS bool, msg,
...
+		return fmt.Errorf("Failed to send test email: %s", err)
 	}
 
+	_, err = w.Write([]byte(msg))
+	if err != nil {
+		return fmt.Errorf("Failed to send test email: %s", err)
+	}
+
+	multipartWriter.Close()
+	wc.Write(buffer.Bytes())
```
Comment (aiwantaozi) [inline pkg/api/customization/alert/notifier_action.go]: is this function has error return?

Comment (alena1108) [reply pkg/api/customization/alert/notifier_action.go]: @zionwu looks like it does, and we should check for the error

Comment (zionwu) [reply pkg/api/customization/alert/notifier_action.go]: sure

**Comment by aiwantaozi** [comment]
LGTM

**Pull request merged**